src/app/clients/add/AddClientForm.tsx

```diff
@@ -183,13 +128,10 @@ const AddClientForm: React.FC<{}> = () => {
         }
 
         try {
-            const ids = await insertClient(clientRecord);
-            try {
-                await insertFamily([...fields.adults, ...fields.children], ids.family_id);
-                router.push("/clients");
-            } catch (error) {
-                await deleteFailedInsert(ids.primary_key);
-                throw error;
+            if (editMode) {
+                await submitFormEditClients(fields, router, clientID);
+            } else {
+                await submitFormAddClients(fields, router);
```

Comment:
I don't think that is necessary as the only difference is where the "if statement" will be in. I personally prefer them to be different functions and have the if statement in the addclientform because that will make it easy to change the specific functions